metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -151,10 +156,33 @@ class WorkspaceLspService(
 
   private val timerProvider: TimerProvider = new TimerProvider(time)
 
+  def getHttpServer(): Future[Option[MetalsHttpServer]] = {
+    httpServer match {
+      // execute client command provider is used to provide the normal doctor
+      case HttpServerOff if !clientConfig.isExecuteClientCommandProvider =>
```

Comment:
Doesn't this mean that if things happen fast enough `startHttpServer` might be called twice? Won't that be problematic?

Reply:
I guess I should also check for http enabled flags, since that will start the server anyway.

Reply:
We should be good nom the logic will only be invoked if httpEnabled is set to false, so it will only happen once now